# Friday Week49
## Work plan
This week, we will be exploring GC3 content of 92 species of soil-dwelling bacteria. We will compare our results to the ones obtained for animals. We will also learn about the effective number of codons (ENC) in bacteria.

## Background
### Codon usage across the tree of life
In mammals, and sexual organisms in general, codon usage is largely determined by the non-adaptive process of GC-biased gene conversion (gBGC), a recombination-associated repair of mismatches favouring GC nucleotides. However, in organisms with high effective population sizes, directional selection can more effectively act to optimize the codon repertoire of genes. This type of selection for codon usage is usually present in fast-growing organisms such as yeast and bacteria, where specific codons confer higher transcription and/or translation efficiency of mRNA. Often, highly expressed genes in bacteria have high translation efficiency that is usually achieved by selecting for codons whose corresponding tRNAs also occur at a high frequency in the cell. Generally, selection for codon usage in microorganisms results in efficient mRNA translation and protein production, and ultimately, faster growth.

### Effective number of codons (ENC)
Codon usage can be quantified by calculating the effective number of codons (ENC; [`Wright, 1990`](https://www.sciencedirect.com/science/article/pii/0378111990904919?via%3Dihub)) on a gene level, or summing over all coding sequences within a genome (i.e., species level). Put simply, the ENC parameter is the number of all the different codons used in a gene to encode the 20 amino acids (assuming that all amino acids can be found within a gene). For example, an extremely biased gene would have an ENC value of 20 such that only one codon is used per amino acid, whereas an unbiased gene would have an ENC value of 61 (i.e., all codons would be in use).

## Strategy

## Questions
